test/export/CodeSystemExporter.test.ts

```diff
@@ -236,6 +236,37 @@ describe('CodeSystemExporter', () => {
     });
   });
 
+  it('should resolve soft indexing when applying Caret Value rules', () => {
+    const codeSystem = new FshCodeSystem('CaretCodeSystem');
+    const contactRule1 = new CaretValueRule('');
+    contactRule1.caretPath = 'contact[+].name';
+    contactRule1.value = 'Example Name';
+    codeSystem.rules.push(contactRule1);
+
+    const contactRule2 = new CaretValueRule('');
+    contactRule2.caretPath = 'contact[=].telecom[+].rank';
```

Comment:
What happens here if the caretPath is instead `contact[=].telecom[=].rank`? That is to say, the `telecom` part of the path is given a soft `=` before any are added with a `+` or a specific number. I would expect that I would get the same result, with the soft `=` being turned into a 0, since even though there aren't any defined elements, this is the first one. Do you think that a warning should be emitted when this is done? Or, do you think that this should be handled as an error?

Reply:
Actually, the way things are set up now, the `=` wouldn't be converted to a number at all without a preceding `+`. I think that's probably something that we'd want to handle as a warning, while still converting the `=` to a 0 without a preceding `+`. I'll ask Chris and/or Mark about it in standup!

Reply:
I agree -- if we run into a `[=]` and there was no preceding `[+]` or `[n]`, then we should convert to `[0]` but issue a warning (since it may be a mistake and should be fixed anyway).